### PR TITLE
add Articles api

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
# What are the relevant tickets?
API for #109 

# Description (What does it do?)
This PR implements a very simple `StaffPost` model and API for creation/editing of rich-text content (e.g., staff writes a highlight of newly released courses). 

# How can this be tested?
1. As a staff user, visit http://localhost:8063/api/v1/staff_posts/
2. POST some data, e.g., `{ "html": "<p>Hello world</p>", "title": "First Post" }`. and then try viewing it at http://localhost:8063/api/v1/staff_posts/POST_ID
3. Try deleting/updating posts, and sending suspicious HTML like `<script>` tags. It should be sanitized.
    - The sanitization is overly aggressive right now, but I'd like to tweak that in a subsequent PR and allow more discussion on https://github.com/mitodl/hq/discussions/2600

### CODE REVIEW

**Comment by Ferdi** [comment]
I'd call it simply "article"  

"staff" assumes predefined permissions , which might change in the future

File: articles/serializers.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+
+@extend_schema_field(str)
+class SanitizedHtmlField(serializers.Field):
+    @staticmethod
+    def to_representation(value):
+        return value
+
+    def to_internal_value(self, data):
+        return nh3.clean(data)
```
Comment (ChristopherChudzicki) [inline articles/serializers.py:15]: See https://github.com/mitodl/hq/discussions/2600 for `nh3` / sanitization discussion.

This is currently too agressive. By default, `nh3.clean(data)` strips classes and data-attributes, which we use for things like embedly cards.

I'd like to configure nh3 more separately from this PR: need experiment more with CKEditor, too, to see exactly what tags we need.

File: open_discussions/views.py [hunk 2]
```diff
@@ -30,6 +30,7 @@ def index(request, **kwargs):  # pylint: disable=unused-argument  # noqa: ARG001
             and (is_admin_user(request) or is_staff_list_editor(request)),
             "is_learning_path_editor": user.is_authenticated
             and (is_admin_user(request) or is_learning_path_editor(request)),
+            "is_article_editor": is_admin_user(request),
```
Comment (ChristopherChudzicki) [inline open_discussions/views.py:33]: I thought about using a separate group like we do for `api/v1/learningpaths/` permissions, but I decided to stay simple until we have a more concrete use case around this.

**Comment by ChristopherChudzicki** [comment]
Closing this in favor of #122

**Pull request closed without merging**